### PR TITLE
[Scaleway] Add module to manage container namespaces

### PR DESCRIPTION
**Pull request by Lunik**
##### SUMMARY

This PR add new modules to manage Scaleway container namespaces services (https://developers.scaleway.com/).

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

- `scaleway_container_namespace`
- `scaleway_container_namespace_info`

##### ADDITIONAL INFORMATION

This PR is a part of the original PR #5359 that was too large.

This PR require evolution on Scaleway config in ansible-test : https://github.com/ansible/ansible/pull/79141

### CODE REVIEW

**Comment by Lunik** [comment]
ready_for_review

File: plugins/modules/cloud/scaleway/scaleway_container_namespace.py [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: scaleway_container_namespace
+short_description: Scaleway Container namespace management module
```
Comment (felixfontein) [inline plugins/modules/cloud/scaleway/scaleway_container_namespace.py:17]:
```suggestion
short_description: Scaleway Container namespace management
```

File: plugins/modules/cloud/scaleway/scaleway_container_namespace.py [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+  environment_variables:
+    description:
+      - Environment variables of the container namespace.
+      - Injected in containers at runtime.
+    type: dict
+
+  secret_environment_variables:
+    description:
+      - Secret environment variables of the container namespace.
+      - Updating thoses values will not output a C(changed) state in Ansible.
```
Comment (felixfontein) [inline plugins/modules/cloud/scaleway/scaleway_container_namespace.py:73]: Same comment as in https://github.com/ansible-collections/community.general/pull/5415#discussion_r1002756112.

From the definition in https://github.com/ansible-collections/community.general/blob/main/plugins/module_utils/scaleway.py#L83-L92 it seems that values from this option are only updated if there are also other changes. This sounds wrong to me.

File: plugins/modules/cloud/scaleway/scaleway_container_namespace_info.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: scaleway_container_namespace_info
+short_description: Scaleway Container namespace info module
```
Comment (felixfontein) [inline plugins/modules/cloud/scaleway/scaleway_container_namespace_info.py:17]:
```suggestion
short_description: Retrieve information on Scaleway Container namespace
```

File: plugins/modules/cloud/scaleway/scaleway_container_namespace.py (hunk 2 above)
Comment (Lunik) [reply plugins/modules/cloud/scaleway/scaleway_container_namespace.py:73]: I agree with you on the topic. I have tried to explain my choice on the matter on the previous PR : https://github.com/ansible-collections/community.general/pull/5399#discussion_r1001579373

tl;dr Some provided field (`secret_environment_variables` in that case) are returned with a different format by the API. There is no way to check if the value must be updated or not.

**Comment by Lunik** [comment]
I have put this PR in `WIP` during the time we try to resolve all issues on #5415

**Comment by Lunik** [comment]
ready_for_review

CI seems to fail because of #5462

**Comment by Lunik** [comment]
ready_for_review

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
If nobody complains, I'll merge this weekend.

**Comment by felixfontein** [comment]
@Lunik thanks for this one as well!
